pkg/library/flatten/merge.go

```diff
@@ -118,3 +120,118 @@ func mergeVolume(into, from *dw.VolumeComponent) error {
 	}
 	return nil
 }
+
+func needsContainerContributionMerge(flattenedSpec *dw.DevWorkspaceTemplateSpec) bool {
+	hasContribution, hasTarget := false, false
+	for _, component := range flattenedSpec.Components {
+		if component.Container == nil {
+			// Ignore attribute on non-container components as it's not clear what this would mean
+			continue
+		}
+		if component.Attributes.GetBoolean(constants.ContainerContributionAttribute, nil) == true {
```

Comment:
This is nitpicking, but maybe an error holder should be added here and at [line 134](https://github.com/devfile/devworkspace-operator/pull/844/commits/0295922d3f69555ddb6cb47d48227d71d9d76dd5#diff-4656ec13493defa6167603ffae8b53cddca10b1b1805a29660dadd8c57f2a3f3R134). It'll make the code that calls this function in `merge.go` a bit uglier though. 

Future calls of `component.Attributes.GetBoolean(constants.ContainerContributionAttribute, nil)` in `mergeContainerContributions` wouldn't need this error holder since  `needsContainerContributionMerge` is called first.

Reply:
This is a good point -- previously I was just interpreting wrong values as false but it makes sense to highlight the case where the attribute _is_ present but cannot be read correctly, e.g. if the value is `ture` or something.

I'll update the PR with that change.